### DISCUSSION TITLE
Multiple classes found for path "User" in the registry of this declarative base

### DISCUSSION THREAD

**Discussion by tonyhart7** [fastapi/fastapi#6476] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "User" in the registry of this declarative base. Please use a fully module-qualified path.
```

### Description

I want to make relationship to my table but this error keep happening 

### Operating System

Windows

### Operating System Details

Windows11

### FastAPI Version

0.70.1

### Python Version

Python 3.10.0

**Comment by insomnes** [comment]
You know it's rly hard to help you with the error only and no code.

**Comment by panla** [comment]
There may be more than one `user (Base)` in the project, and it belonging to Sqlalchemy.

**Comment by tonyhart7** [comment]
> You know it's rly hard to help you with the error only and no code.

the thing is exception from sqlalchemy not from my code

**Comment by tonyhart7** [comment]
> There may be more than one `user (Base)` in the project, and it belonging to Sqlalchemy.

that the strange part, its only 1 class User in my code

**Comment by insomnes** [comment]
> the thing is exception from sqlalchemy not from my code

Yeah, it is. But without any reproducible code it's very hard to say what causes it.

**Comment by tonyhart7** [comment]
> > the thing is exception from sqlalchemy not from my code
> 
> Yeah, it is. But without any reproducible code it's very hard to say what causes it.

~~~
class Post(Base):
    __tablename__ = "posts"
    __table_args__ = {'extend_existing': True}
    
    id = Column(Integer, primary_key=True, nullable=False)
    title = Column(String, nullable=False)
    content = Column(String, nullable=False)
    published = Column(Boolean, server_default='TRUE', nullable=False)
    created_at = Column(TIMESTAMP(timezone=True),
                        nullable=False, server_default=text('now()'))
    owner_id = Column(Integer, ForeignKey(
        "users.id", ondelete="CASCADE"), nullable=False)

    owner = relationship("User")

class User(Base):
    __tablename__ = "users"
    __table_args__ = {'extend_existing': True}
    
    id = Column(Integer, primary_key=True, nullable=False)
    email = Column(String, nullable=False, unique=True)
    password = Column(String, nullable=False)
    created_at = Column(TIMESTAMP(timezone=True),
                        nullable=False, server_default=text('now()'))
~~~

**Comment by xdewx** [comment]
> > > the thing is exception from sqlalchemy not from my code
> > 
> > 
> > Yeah, it is. But without any reproducible code it's very hard to say what causes it.
> 
> ```
> class Post(Base):
>     __tablename__ = "posts"
>     __table_args__ = {'extend_existing': True}
>     
>     id = Column(Integer, primary_key=True, nullable=False)
>     title = Column(String, nullable=False)
>     content = Column(String, nullable=False)
>     published = Column(Boolean, server_default='TRUE', nullable=False)
>     created_at = Column(TIMESTAMP(timezone=True),
>                         nullable=False, server_default=text('now()'))
>     owner_id = Column(Integer, ForeignKey(
>         "users.id", ondelete="CASCADE"), nullable=False)
> 
>     owner = relationship("User")
> 
> class User(Base):
>     __tablename__ = "users"
>     __table_args__ = {'extend_existing': True}
>     
>     id = Column(Integer, primary_key=True, nullable=False)
>     email = Column(String, nullable=False, unique=True)
>     password = Column(String, nullable=False)
>     created_at = Column(TIMESTAMP(timezone=True),
>                         nullable=False, server_default=text('now()'))
> ```

hey, i came across the same question, how did you solve it?

**Comment by Chibuoyimm** [comment]
in the bracket where you define the class name, embed the path to the class. for example, if your models file is in a folder, you write something like: owner = relationship("folder_name.file_name.class_name"). hope this helps!
